spring-cloud-kubernetes-fabric8-discovery/src/main/java/org/springframework/cloud/kubernetes/fabric8/discovery/KubernetesDiscoveryClient.java

```diff
@@ -146,17 +146,9 @@ private List<ServiceInstance> getNamespaceServiceInstances(EndpointSubsetNS es,
 			int endpointPort = endpointsPort(endpointSubset, serviceId, properties, service);
 			List<EndpointAddress> addresses = addresses(endpointSubset, properties);
 			for (EndpointAddress endpointAddress : addresses) {
-
```

Comment:
I removed this code and moved it to the utils + added a test to assert proper fields. 

I've also added the namespace as part of the result.